ram.tf

```
@@ -1,3 +1,14 @@
+locals {
+  ram_principals_provided = var.ram_principal != null || var.ram_principals != []
```

this will not work 

```
var.ram_principals != []
```

TF comparison operators don't compare lists.

```suggestion
  ram_principals_provided = var.ram_principal != null || length(var.ram_principals) > 0
```

or if we want to check for `null`

```
  ram_principals_provided = var.ram_principal != null || try(length(var.ram_principals), 0) > 0
```

---

thank you, I went with your first suggestion